src/client/datascience/constants.ts

```diff
@@ -18,18 +18,31 @@ export const KernelLauncherDaemonModule = 'vscode_datascience_helpers.kernel_lau
 export const PythonExtension = 'ms-python.python';
 
 // List of 'language' names that we know about. All should be lower case as that's how we compare.
+export const KnownKernelLanguageAliases = new Map<string, string>([
+    ['qsharp', 'q#'],
+    ['csharp', 'c#'],
+    ['fsharp', 'f#'],
+    ['c++11', 'c++'],
+    ['c++12', 'c++'],
+    ['c++14', 'c++']
+]);
 export const KnownNotebookLanguages: string[] = [
     'python',
     'r',
     'julia',
     'c++',
     'c#',
     'f#',
+    'q#',
+    'powershell',
+    'java',
```

Comment:
Added missing `rust, java & powershell`.
Recently we assumed that no one uses java, but we didn't track it.